kube-controllers/tests/fv/metrics_test.go

```diff
@@ -448,3 +565,91 @@ func getMetrics(metricsURL string) (string, error) {
 	}
 	return string(body), nil
 }
+
+func createNode(node string, k8sClient *kubernetes.Clientset) {
+	_, err := k8sClient.CoreV1().Nodes().Create(context.Background(),
+		&v1.Node{
+			TypeMeta:   metav1.TypeMeta{Kind: "Node", APIVersion: "v1"},
+			ObjectMeta: metav1.ObjectMeta{Name: node},
+			Spec:       v1.NodeSpec{},
+		},
+		metav1.CreateOptions{})
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+}
+
+func createIPPool(name string, cidr string, calicoClient client.Interface) {
+	p := api.NewIPPool()
+	p.Name = name
+	p.Spec.CIDR = cidr
+	p.Spec.BlockSize = 26
+	p.Spec.NodeSelector = "all()"
+	p.Spec.Disabled = false
+	_, err := calicoClient.IPPools().Create(context.Background(), p, options.SetOptions{})
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+}
+
+func allocatePodIpWithHandle(ip string, handle string, node string, pod string, calicoClient client.Interface) {
+	attrs := map[string]string{"node": node, "pod": pod, "namespace": "default"}
+	err := calicoClient.IPAM().AssignIP(context.Background(), ipam.AssignIPArgs{
+		IP: net.MustParseIP(ip), HandleID: &handle, Attrs: attrs, Hostname: node,
+	})
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
```

Comment:
Hm, for most calls an offset of 1 is correct, but we do also call this from within `createPod` which would probably like to have an offset of 2 given it's two layers deep in the call stack from the test.

Reply:
Nice catch! Fixed with my latest changes